src/Core/Settings.h

```diff
@@ -80,9 +80,14 @@ class IColumn;
     M(Bool, use_uncompressed_cache, false, "Whether to use the cache of uncompressed blocks.", 0) \
     M(Bool, replace_running_query, false, "Whether the running request should be canceled with the same id as the new one.", 0) \
     M(UInt64, background_buffer_flush_schedule_pool_size, 16, "Number of threads performing background flush for tables with Buffer engine. Only has meaning at server startup.", 0) \
-    M(UInt64, background_pool_size, 16, "Number of threads performing background work for tables (for example, merging in merge tree). Only has meaning at server startup.", 0) \
+    M(UInt64, background_pool_size, 16, "Deprecated???", 0) \
```

Comment:
Not sure what to do with it.
Maybe set it to zero by default, but in case it is specified in config, use it for `background_merges_pool_size` and `background_mutations_pool_size`?

Reply:
Maybe throw some kind of exception?

Reply:
Maybe ... but it would be more difficult to update server

Reply:
Also the same is for `number_of_free_entries_in_pool_to_lower_max_size_of_merge`

Reply:
Keep it and limit merges + mutations to max of that? This way the server could be upgraded and settings tuned later. Also make merges and mutations pool size = to this setting by default.

We have this setting configured to much higher values on systems with good disks/high cpu count.

As I see it the above proposal is the best for easy upgrades. Otherwise we'd have to somehow configure settings prior to upgrade, but that can't be done as older server wouldn't be able to start anymore due to unknown settings...

tldr; `background_merges_pool_size = background_mutations_pool_size = background_pool_size` and `max merges + max mutations = background_pool_size`

ideally...